state/protocol/events.go

```diff
@@ -84,10 +84,24 @@ type Consumer interface {
 	// NOTE: Only called once the phase transition has been finalized.
 	EpochCommittedPhaseStarted(currentEpochCounter uint64, first *flow.Header)
 
-	// EpochEmergencyFallbackTriggered is called when epoch fallback mode [EFM] is triggered.
-	// Since EFM is a permanent, spork-scoped state, this event is triggered only once.
-	// After this event is triggered, no further epoch transitions will occur,
-	// no further epoch phase transitions will occur, and no further epoch-related
-	// related protocol events (the events defined in this interface) will be emitted.
-	EpochEmergencyFallbackTriggered()
+	// EpochFallbackModeTriggered is called when epoch fallback mode [EFM] is triggered.
+	// EFM is triggered when an invalid or unexpected epoch-related service event is observed,
+	// or an expected service event is not observed before the epoch commitment deadline.
+	// After EFM is triggered, we do not transition into any not-yet-committed epoch until
+	// an EpochRecover event is observed, after which regular epoch transitions begins again.
+	// Usually, this means we remain in the current epoch until EFM is exited.
+	// If EFM was triggered within the EpochCommitted phase, then we complete the transition
+	// to the next, already-committed epoch, then remain in that epoch until EFM is exited.
+	// NOTE: Only called once the block triggering EFM is finalized.
+	EpochFallbackModeTriggered()
+
+	// EpochFallbackModeExited is called when epoch fallback mode [EFM] is exited.
+	// EFM is exited when an EpochRecover service event is processed, which defines
+	// a final view for the current epoch and fully specifies the subsequent epoch.
+	// NOTE: Only called once the block incorporating the EpochRecover is finalized.
+	EpochFallbackModeExited()
+
+	// EpochExtended is called when a flow.EpochExtension is added to the current epoch
+	// NOTE: Only called when the block including the new extension is finalized.
```

Comment:
I find this wording ambiguous, because "contained in block" usually means "included in payload" (like collections, seals). Observing a block past the threshold triggers the state machine to insert the extension into the state. 

```suggestion
	// NOTE: This notification is emitted when the block triggering the EFM extension is finalized.
```

Reply:
> Observing a block past the threshold triggers the state machine to insert the extension into the state.

"... into the state", which is committed to in the block payload.

Using "included" language here because the state is included in the block payload (just in a different way). It is added via the protocol state machine rather than by pulling from mempools, and by reference (hash commitment) rather than by copy (direct embedding), but I still think "included/contained" is useful terminology for objects in the protocol state.